shared/src/main/scala/ListHelpers.scala

```diff
@@ -5,6 +5,27 @@ import VNum.given
 
 object ListHelpers:
 
+  def filter(iterable: VList, predicate: VFun)(using ctx: Context): VList =
+    val branches = predicate.originalAST match
+      case Some(lam) => lam.body
+      case None =>
+        return VList(iterable.zipWithIndex.map { (item, index) =>
```

Comment:
The `return` here confused me a bit, I thought you were setting branches to `VList(iterable.zipWith...` at first. I think all the code below (`val filtered = ...`) can go into the `Some` branch above